support-frontend/app/services/stepfunctions/SupportWorkersClient.scala

```
@@ -105,64 +108,78 @@ class SupportWorkersClient(
 
-    val createPaymentMethodState = CreatePaymentMethodState(
-      requestId = requestId,
-      user = user,
-      giftRecipient = getGiftRecipient(request.body),
-      product = request.body.product,
-      paymentProvider = PaymentProvider.fromPaymentFields(request.body.paymentFields.left.toOption),
-      paymentFields = request.body.paymentFields,
-      acquisitionData = Some(AcquisitionData(
-        ophanIds = request.body.ophanIds,
-        referrerAcquisitionData = referrerAcquisitionDataWithGAFields(request),
-        supportAbTests = request.body.supportAbTests
-      )),
-      promoCode = request.body.promoCode,
-      firstDeliveryDate = request.body.firstDeliveryDate
-    )
-    val isExistingAccount = createPaymentMethodState.paymentFields.left.exists(_.isInstanceOf[ExistingPaymentFields])
-    underlying.triggerExecution(createPaymentMethodState, user.isTestUser, isExistingAccount).bimap(
-      { error =>
-        SafeLogger.error(scrub"[$requestId] Failed to trigger Step Function execution for ${user.id} - $error")
-        StateMachineFailure: SupportWorkersError
-      },
-      { success =>
-        SafeLogger.info(s"[$requestId] Successfully triggered Step Function execution for ${user.id} ($success)")
-        underlying.jobIdFromArn(success.arn).map { jobId =>
-          StatusResponse(
-            status = Status.Pending,
-            trackingUri = supportUrl + statusCall(jobId).url
-          )
-        } getOrElse {
-          SafeLogger.error(scrub"[$requestId] Failed to parse ${success.arn} to a jobId after triggering Step Function execution for ${user.id} $request")
-          StatusResponse(
-            status = Status.Failure,
-            trackingUri = "",
-            failureReason = Some(CheckoutFailureReasons.Unknown)
-          )
+    for {
+      giftRecipient <- EitherT.fromEither[Future](request.body.giftRecipient.map(getGiftRecipient(_, request.body.product)).sequence)
```

the key change here is that it's now an error to have inconsistent giftee details from the client side, rather than just silently continuing as a non-gift subscription

---

Thanks for flagging this JD - I'll change this to the new model on clientside 👍

---

Or actually, it's probably best if I leave it as you've handled the changes in submit.js and we just make sure it works end-to-end